### PR TITLE
New tasks patches

### PR DESCRIPTION
**Pull request by valeriadamante**
This PR is intended to
1. add patches to the previous PR 
2. resolve conflicts with k's PR
3. unify the hist merge + hadd merged task
4. remove ext1

### CODE REVIEW

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: Analysis/tasks.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import law
 import os
 import yaml
+import threading
```
Comment (kandrosov) [inline Analysis/tasks.py:4]:
```suggestion
```

**Review by kandrosov — CHANGES_REQUESTED** [review]

File: config/HH_bbtautau/global.yaml [hunk 2]
```diff
@@ -50,4 +50,232 @@ categories:
...
+  res1b: Medium
+  boosted: Loose
+
+mass_cut_limits:
+  bb_m_vis:
+    - 50
+    - 270
+  tautau_m_vis:
+    - 20
+    - 130
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply config/HH_bbtautau/global.yaml:99]: yes. I obtained:
```
quantile max (95%) for bb mass resolved =  144.77955932617186
quantile min (95%) for bb mass resolved =  75.6971378326416
quantile max (95%)  for tautau mass resolved =  106.78509750366207
quantile min (95%)  for tautau mass resolved =  35.552077293395996

quantile max (99.5%) for bb mass resolved =  267.4022723388703
quantile min (99.5%) for bb mass resolved =  46.51988693237305
quantile max (99.5%) for tautau mass resolved =  121.6530269622805
quantile min (99.5%) for tautau mass resolved =  23.111099338531496
```

and we went for 20, 130 for tauTau_m_vis

File: config/HH_bbtautau/global.yaml [hunk 3]
```diff
@@ -50,4 +50,232 @@ categories:
...
+  - bTagShapeSF_lfstats2
+  - bTagShapeSF_hfstats1
+  - bTagShapeSF_hfstats2
+  - bTagShapeSF_cferr1
+  - bTagShapeSF_cferr2
+
+vars_to_plot:
+    - tautau_m_vis
+    - bb_m_vis
+    - { name: kinFit_m, need_cache: true }
```
[the remark this answers is not in the document]
Comment (valeriadamante) [reply config/HH_bbtautau/global.yaml:281]: ok

File: Analysis/tasks.py (hunk 1 above)
Comment (valeriadamante) [reply Analysis/tasks.py:4]: ok

File: Analysis/HistMerger.py [hunk 4]
```diff
@@ -242,26 +204,21 @@ def GetBTagWeightDict(var, all_histograms):
...
     MergeHistogramsPerType(all_histograms)
     all_histograms_1D=GetBTagWeightDict(args.var,all_histograms)
-    fixNegativeContributions=True
-    if args.var == 'kinFit_m':
-        fixNegativeContributions=False
-
-    AddQCDInHistDict(args.var,all_histograms_1D, channels, categories, args.uncSource, all_samples_types.keys(), scales,fixNegativeContributions)
+    fixNegativeContributions = False
+    if args.var != 'kinFit_m':
+        fixNegativeContributions=True
```
Comment (kandrosov) [inline Analysis/HistMerger.py:211]: take this from config (same as 'need_cache')

**Review by kandrosov — CHANGES_REQUESTED** [review]

**Review by kandrosov — APPROVED** [review]

**Pull request merged**